### PR TITLE
refactor: dbShutdown and add event handlers for it

### PR DESCRIPTION
**Pull request by rodrigo4244**
This PR refactor dbShutdown to use async await and add event handlers when the connection is disconnecting and when it was successfully disconnected.

### CODE REVIEW

File: config/index.js [hunk 1]
```diff
@@ -52,15 +52,19 @@ const initConnection = (opts = {}) => {
   dbConnection = mongoose.connection
   dbConnection.on('error', handleErr)
   dbConnection.on('open', () => { resolveConnection(dbConnection) })
+  dbConnection.on('disconnecting', () => logger.info('disconnecting mongodb...'))
+  dbConnection.on('disconnected', () => logger.info('mongodb connection successfully disconnected.'))
 }
 
-const dbShutdown = cb => {
+const dbShutdown = async cb => {
```
Comment (gunar) [inline config/index.js]: Guess you can get rid of `cb`

File: config/index.js [hunk 2]
```diff
@@ -52,15 +52,19 @@ const initConnection = (opts = {}) => {
...
   logger.info('Shutting down db connection')
   // This may be a sudden termination and not wait for all saves to finish
-  dbConnection.close(err => {
+  try {
+    await dbConnection.close()
+  } catch (err) {
     logger.error(err)
     mongoose.disconnect(cb)
```
Comment (gunar) [inline config/index.js]: Does this work?
```js
return mongoose.disconnect()
```

**Comment by rodrigo4244** [comment]
After your review I would squash the commits and put the change on `CHANGELOG` in a separate commit.

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1 @@
+- Refactor dbShutdown and add event handlers for it.
```
Comment (gunar) [inline CHANGELOG.md]: Packages don't need CHANGELOG, just products (so that non-tech people know what is going on).

File: config/index.js [hunk 4]
```diff
@@ -40,27 +40,30 @@ const defaultOptions = {
...
-const dbShutdown = cb => {
+const dbShutdown = async () => {
   logger.info('Shutting down db connection')
   // This may be a sudden termination and not wait for all saves to finish
-  dbConnection.close(err => {
+  try {
+    await dbConnection.close()
+  } catch (err) {
     logger.error(err)
-    mongoose.disconnect(cb)
```
Comment (gunar) [inline config/index.js]: I think you accidentally deleted `mongoose.disconnect(cb)`

Friendly reminder: always review your own PRs before sending for review.

**Pull request merged**